lib/note-content-editor.jsx

```diff
@@ -1,11 +1,68 @@
 import React, { PropTypes } from 'react';
-import { Editor, EditorState, ContentState } from 'draft-js';
+import {
+	ContentState,
+	Editor,
+	EditorState,
+	Modifier,
+} from 'draft-js';
 import { invoke, noop } from 'lodash';
 
 function plainTextContent( editorState ) {
 	return editorState.getCurrentContent().getPlainText( '\n' )
 }
 
+// contains methods that mirror the methods on Modifier with 2 differences:
+// 1. operate on EditorState instances
+// 2. move selection to the natural place following the modification
+const EditorStateModifier = {
+	insertText( editorState, selection, characters ) {
+		// this function assumes selection is collapsed
+
+		const afterInsert = EditorState.push(
+			editorState,
+			Modifier.insertText(
+				editorState.getCurrentContent(), selection, characters
+			),
+			'insert-characters'
+		);
+
+		const prevSelection = editorState.getSelection();
+		const prevOffset = prevSelection.getStartOffset();
+		const offsetDiff = characters.length;
+
+		// move selection forward by nr of chars that were added
+		const nextSelection = prevSelection.merge( {
+			anchorOffset: prevOffset + offsetDiff,
+			focusOffset: prevOffset + offsetDiff,
+		} )
+
+		return EditorState.forceSelection( afterInsert, nextSelection );
+	},
+
+	removeRange( editorState, selection = editorState.getSelection() ) {
+		// this function assumes startKey and endKey are the same
+		// ie. selection doesn't span more than one block
```

Comment:
this one assumes this because my uses of it so far never need to operate on ranges that span across lines